pkg/provisioner/ironic/testserver/ironic.go

```diff
@@ -75,3 +87,45 @@ func (m *IronicMock) WithNodeStatesProvision(nodeUUID string) *IronicMock {
 	m.ResponseWithCode("/v1/nodes/"+nodeUUID+"/states/provision", "{}", http.StatusAccepted)
 	return m
 }
+
+// NoNode configures the server so /v1/nodes/name returns a 404
+func (m *IronicMock) NoNode(name string) *IronicMock {
+	m.NotFound(fmt.Sprintf("/v1/nodes/%s", name))
+	return m
+}
+
+// CreateNodes configures the server so POSTing to /v1/nodes saves the data
+func (m *IronicMock) CreateNodes() *IronicMock {
+	m.Handler("/v1/nodes", func(w http.ResponseWriter, r *http.Request) {
+		bodyRaw, err := ioutil.ReadAll(r.Body)
+		if err != nil {
+			m.logRequest(r, fmt.Sprintf("ERROR: %s", err))
+			http.Error(w, fmt.Sprintf("%s", err), 500)
+			return
+		}
+
+		body := string(bodyRaw)
+		m.t.Logf("%s: create nodes request %v", m.name, body)
+
+		// The UUID value doesn't actually have to be a UUID, so we
+		// just make a new string based on the count of nodes already
+		// created.
+		uuid := fmt.Sprintf("node-%d", len(m.CreatedNodes))
+		m.t.Logf("%s: uuid %s", m.name, uuid)
+
+		// Record what we have so the test can examine it later
+		m.CreatedNodes = append(m.CreatedNodes, CreatedNode{
+			Body: body,
+			UUID: uuid,
+		})
+
+		// hackily add uuid to the json response by inserting it to the front of the string
+		response := fmt.Sprintf("{\"uuid\": \"%s\", %s", uuid, strings.TrimLeft(body, "{"))
```

Comment:
Just as a confirmation, is it fine to return all the fields sent in the  `body` request as a response? From a quick glance on the API [here](https://docs.openstack.org/api-ref/baremetal/?expanded=create-node-detail) should be fine. Maybe in future we could need something more sophisticated, but for the current test seems ok